src/integration.ts

```diff
@@ -261,31 +270,31 @@ export default class Integration {
     }
 
     try {
-      const params = {
-        RoleName: `NewRelicLambdaIntegrationRole_${accountId}`
-      };
-
-      const {
-        Role: { Arn }
-      } = await this.awsProvider.request("IAM", "getRole", params);
-
-      return Arn;
+      return this.requestRoleArn(`NewRelicLambdaIntegrationRole_${accountId}`);
     } catch (err) {
-      this.serverless.cli.log(
-        "The required NewRelicLambdaIntegrationRole cannot be found; Creating Stack with NewRelicLambdaIntegrationRole."
-      );
-      const stackId = await this.createCFStack(accountId);
-      waitForStatus(
-        {
-          awsMethod: "describeStacks",
-          callbackMethod: () => this.enable(externalId),
-          methodParams: {
-            StackName: stackId
+      // We're limited to one rolename and a 64-character regex, so allowing for streams means a second query
+      try {
+        return this.requestRoleArn(
+          `NewRelicInfrastructure-Integrations|NewRelicStagingIntegration`
```

Comment:
Is this always going to have one of these values? Is there a more definitive way to check for a metric stream, such as via NerdGraph or something like that?

Reply:
There's nothing to prevent someone from creating an arbitrary name for the integration. The default suggested name for the streams integration is `NewRelicInfrastructure-Integrations`, and this also captures our staging integration.

While someone could create an integration with an arbitrary name...this AWS-side check happens *after* a nerdgraph query that would reveal an integration, no matter how it was named. If we find an integration there, we don't run this. As a result, this check would be a seldom-triggered backstop.

Reply:
Digging in a bit more, AWS throws an error when we query it with the wildcards, so this multiple-check is still needed, if NerdGraph doesn't return existing integrations.